lib/recipe.rb

```diff
@@ -0,0 +1,21 @@
+class Recipe
+  attr_reader:name, :ingredients_required, :ingredients
+  def initialize(name)
+    @name = name
+    @ingredients_required = Hash.new(0)
+    @ingredients = []
```

Comment:
I'd rather this be a method that dynamically calculates the ingredients based off of the ingredients_required hash, that way we don't have two sources of truth, but the way you coded this does work.